clients/python-wrapper/lakefs/branch.py

```diff
@@ -24,9 +23,13 @@
     TransactionException
 )
 
-# Unless stated otherwise by passing `-W <something>` to Python, we display `DeprecationWarning`s by default.
-if not sys.warnoptions:
-    warnings.simplefilter('always', DeprecationWarning)
+
+class LakeFSDeprecationWarning(Warning):
+    """
+    This implementation is a replacement for the standard `DeprecationWarning` class,
+    because the `DeprecationWarning` class is ignored by default by the `warnings` module.
```

Comment:
The pydoc for DeprecationWarning is
> Base class for warnings about deprecated features.

So I would propose:
```suggestion
    Warning about use of a deprecated lakeFS or client feature.  Unlike
    `DeprecationWarning`, this class is displayed by default.  See
    <https://docs.python.org/3/library/warnings.html#default-warning-filter>
    for how to disable it.
```

This uses a more positive and active voice in documentation (**what** it _is_ rather than **why** it _isn't_ something else).